frontend/views/components/modal/Modal.vue

```diff
@@ -154,6 +158,11 @@ export default ({
     },
     setModalQueries (componentName, queries) {
       this.queries[componentName] = queries
+    },
+    isPreauthGuarded (toModal) {
```

Comment:
While investigating this issue, noticed that we are able to open `GroupCreationModal.vue` even when the user is not logged in yet, when it's not supposed to. So worked on adding a guard logic in `$route` watcher of `Modal.vue` here. (Pls let me know if any of you think there is a better way of achieving this..!)

Reply:
Can't this be achieved by adding a guard in `~/frontend/controller/router.js`?